#! /bin/sh
patch -p0 -f $* < $0
exit $?

2008-11-18  Werner Koch  <wk@g10code.com>

	* decrypt.c (_gpgme_decrypt_status_handler): Use
	_gpgme_map_gnupg_error to parse the error code for decrypt.algorithm.


Index: gpgme/decrypt.c
===================================================================
--- gpgme/decrypt.c   (revision 1345)
+++ gpgme/decrypt.c   (working copy)
@@ -180,7 +180,6 @@
          related to the backend.  */
       {
        const char d_alg[] = "decrypt.algorithm";
-       const char u_alg[] = "Unsupported_Algorithm";
        const char k_alg[] = "decrypt.keyusage";
 
        if (!strncmp (args, d_alg, sizeof (d_alg) - 1))
@@ -189,11 +188,13 @@
            while (*args == ' ')
              args++;
 
-           if (!strncmp (args, u_alg, sizeof (u_alg) - 1))
+           if (gpg_err_code (_gpgme_map_gnupg_error (args))
+                == GPG_ERR_UNSUPPORTED_ALGORITHM)
              {
                char *end;
 
-               args += sizeof (u_alg) - 1;
+               while (*args && *args != ' ')
+                 args++;
                while (*args == ' ')
                  args++;
 



